### PR TITLE
fix long startup delays in scripts/unittest

### PR DESCRIPTION
**Pull request by jsteemann**
### Scope & Purpose

Fix startup delays in arangosh in case there is no arangod instance running.
Additionally, adjust `scripts/unittest` to not try to connect to a running instance with its arangosh, but use `none` as server endpoint.
Furthermore, add configurable delays for connection retry pauses and expose the connection builder's configuration setters so arangosh can use them.

- [x] Bug-Fix for *devel-branch* (i.e. no need for backports?)
- [x] The behavior in this PR can be (and was) *manually tested* (support / qa / customers can test it)
- [x] The behavior change can be verified via automatic tests

### Testing & Verification

This change is already covered by existing tests, such as *scripts/unittest*. Existing tests need to continue to work with this adjustment.

http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8268/

### CODE REVIEW

File: 3rdParty/fuerte/src/GeneralConnection.cpp [hunk 1]
```diff
@@ -110,30 +110,35 @@ void GeneralConnection<ST>::tryConnect(unsigned retries) {
       }
     });
   } else {
-    asio_ns::error_code ec;
-    _proto.timer.cancel(ec);
+    _proto.timer.cancel();
```
Comment (graetzer) [inline 3rdParty/fuerte/src/GeneralConnection.cpp]: this may throw now

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8291/

File: 3rdParty/fuerte/src/GeneralConnection.cpp (hunk 1 above)
Comment (jsteemann) [reply 3rdParty/fuerte/src/GeneralConnection.cpp]: I reverted that change now

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8295/

File: 3rdParty/fuerte/src/GeneralConnection.cpp [hunk 2]
```diff
@@ -112,10 +127,12 @@ void GeneralConnection<ST>::tryConnect(unsigned retries) {
...
   FUERTE_ASSERT(_state.load() == Connection::State::Connecting);
   FUERTE_LOG_DEBUG << "tryConnect (" << retries << ") this=" << this << "\n";
-  
+
   auto self = Connection::shared_from_this();
-  if (_config._connectTimeout.count() > 0) {
-    _proto.timer.expires_after(_config._connectTimeout);
+  auto timeout = relativeTimeout(_config, start);
+
+  if (timeout.count() > 0) {
```
Comment (graetzer) [inline 3rdParty/fuerte/src/GeneralConnection.cpp]: isn't this always > 0 now

Comment (jsteemann) [reply 3rdParty/fuerte/src/GeneralConnection.cpp]: true

**Comment by jsteemann** [comment]
http://172.16.10.101:8080/view/PR/job/arangodb-matrix-pr/8308/

**Pull request merged**